ChangeLog.md

```diff
@@ -1,11 +1,105 @@
 
 # Change Log
 
-## Version 10.XX.XX
+## Version 10.3.0
 
-### Breaking change
+### Breaking changes
+
+1. (TBC based on outcome of backslash escape vs --strip-top-dir)  The `*` character is no longer supported as a wildcard in URLs.  (It
+   remains supported in local file paths.)
+   1. The one execption is that `/*` is still allowed at the very end of the "path" section of a
+      URL. This is illustrated by the difference between these two source URLs:
+      `https://account/container/virtual?SAS` and 
+      `https://account/container/virtualDir/*?SAS`.  The former copies the virtual directory
+      `virtualDir` by creating a folder of that name at the destination.  The latter copies the
+      _contents_ of `virtual` dir directly into the target without creating a folder named "virtualDir".
+   1. If you need to refer a literal `*` in the name of a blob, e.g. for a blob named "*", escape it with a backslash as `\*`
+1. The `--include` and `--exclude` parameters have been replaced by `--include-pattern` and
+   `--exclude-pattern` (for filenames) and `--include-path` and `--exclude-path` (for paths,
+   including directory and filenames).
+   The new parameters have behaviour that is better defined in complex situations (such as
+   recursion).  The `*` wildcard is supported in the pattern parameters, but _not_ in the path ones.
+1. There have been two breaking changes to the JSON output that is produced if you request
+   JSON-formatted output. The `sync` command's output in JSON has changed for consistency reasons,
+   and the final message type, for `copy` and `sync` has changed its name from `Exit` to `EndOfJob`.
+   Tools using the JSON output format to integrate AzCopy should be aware.
+1. If downloading to "null" on Windows the target must now be named "NUL", according to standard
+   Windows conventions.  "/dev/null" remains correct on Linux. (This feature can be used to test
+   throughput or check MD5s without saving the downloaded data.) 
+1. The file format of the (still undocmented) `--list-of-files` parameter is changed.  (It remains
```

Comment:
TODO: Note how this has changed. I understand it's a undocumented parameter designed for ASE, but we don't know for sure if there's anybody else out there using it.

Reply:
Good point. Generally, its better if the app can warn folks about changes, rather than relying on them to read the docs. (Because folks are notoriously casual about reading technical docs).   I've just added a PR to implement such a warning.

Also, I've already put a small update into the wiki page: https://github.com/Azure/azure-storage-azcopy/wiki/Listing-specific-files-to-transfer

Reply:
Good stuff.

---
In reply to: [333358972](https://github.com/Azure/azure-storage-azcopy/pull/607#discussion_r333358972) [](ancestors = 333358972)